src/sdk/accountspace_fs.js

```diff
@@ -259,73 +231,233 @@ class AccountSpaceFS {
+                secret_key: generated_secret_key,
+                creation_date: new Date().toISOString(),
+                deactivated: false,
+            };
+            const requested_account_encrypted = await nc_mkm.encrypt_access_keys(requested_account);
+            const account_to_create_access_keys_string = JSON.stringify(requested_account_encrypted);
+            nsfs_schema_utils.validate_account_schema(JSON.parse(account_to_create_access_keys_string));
+            await native_fs_utils.update_config_file(this.fs_context, this.accounts_dir,
+                requested_account_config_path, account_to_create_access_keys_string);
+            await this._create_access_key_symlink(requested_account.name, generated_access_key);
+            return {
+                username: requested_account.name,
+                access_key: requested_account.access_keys[index_for_access_key].access_key,
+                create_date: requested_account.access_keys[index_for_access_key].creation_date,
+                status: this._get_access_key_status(requested_account.access_keys[index_for_access_key].deactivated),
+                secret_key: generated_secret_key,
+            };
+        } catch (err) {
+            dbg.error(`AccountSpaceFS.${action} error`, err);
+            throw this._translate_error_codes(err, entity_enum.ACCESS_KEY);
+        }
     }
 
+    // 1 - read the symlink file that we get in params (access key id)
+    // 2 - check if the access key that was received in param exists
+    // 3 - read the config file
+    // 4 - check that config file is on the same root account
+    // General note: only serves the requester (no flag --user-name is passed)
     async get_access_key_last_used(params, account_sdk) {
-        dbg.log1('get_access_key_last_used', params);
-        return {
-            region: dummy_region,
-            last_used_date: new Date(Date.now() - 30 * MS_PER_MINUTE),
-            service_name: dummy_service_name,
-            username: dummy_user1.username,
-        };
+        const action = 'get_access_key_last_used';
+        dbg.log1(`AccountSpaceFS.${action}`, params, account_sdk);
+        try {
+            const requesting_account = account_sdk.requesting_account;
+            const access_key_id = params.access_key;
+            const requested_account_path = get_symlink_config_file_path(this.access_keys_dir, access_key_id);
+            await this._check_if_account_exists_by_access_key_symlink(action, requesting_account, requested_account_path, access_key_id);
+            const requested_account = await get_config_data(this.config_root_backend, requested_account_path, true);
+            this._check_if_requested_account_same_root_account_as_requesting_account(action,
+                requesting_account, requested_account);
+            return {
+                region: dummy_region, // GAP
+                last_used_date: new Date(Date.now() - 30 * MS_PER_MINUTE), // GAP
```

Comment:
why not just new Date() for simplicity? is there any need to make this date appear in the past??